src/e3/python/pypi.py

```diff
@@ -365,6 +377,7 @@ def __init__(
         cache_file: str | None = None,
         pypi_url: str = "https://pypi.org/pypi",
         allowed_prerelease: list[str] | None = None,
+        allowed_yanked: list[str] | None = None,
```

Comment:
You should add the corresponding DocString for this parameter, allowed_prerelease is also missing